votingsystem/src/general_meeting.rs

```diff
@@ -0,0 +1,262 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pub use ctypes::{BlockId};
+use coordinator::context::SubStorageAccess;
+use coordinator::types::*;
+use parking_lot::RwLock;
+use coordinator::module::*;
+use crate::common::*;
+use foundry_module_rt::UserModule;
+use serde::{Deserialize, Serialize};
+use remote_trait_object::raw_exchange::{import_service_from_handle, HandleToExchange, Skeleton};
+use remote_trait_object::{service, Context as RtoContext, Service, ServiceRef};
+use rand::Rng;
+use std::sync::Arc;
+
+
+enum ExecuteError {
+    InvalidMetadata,
+    InvalidSign,
+    InvalidFormat,
+}
+
+struct Context{
+    pub storage: Option<Box<dyn SubStorageAccess>>,
+    pub meeting: Option<Box<dyn MeetingManager>>,
+}
+
+impl Context {
+    fn storage(&self) -> &dyn SubStorageAccess {
+        self.storage.as_ref().unwrap().as_ref()
+    }
+
+    fn storage_mut(&mut self) -> &mut dyn SubStorageAccess {
+        self.storage.as_mut().unwrap().as_mut()
+    }
+
+    fn meeting(&self) ->&dyn MeetingManager {
+        self.meeting.as_ref().unwrap().as_ref()
+    }
+    fn meeting_mut(&mut self)->&mut dyn MeetingManager {
+        self.meeting.as_mut().unwrap().as_mut()
+    }
+}
+
+impl Service for Context {}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TxGeneralMeeting{
+    pub end_time: TimeStamp,
+    pub tallying_time: TimeStamp,
+}
+
+impl Action for TxGeneralMeeting{}
+pub type OwnTransaction = crate::common::SignedTransaction<TxGeneralMeeting>;
+
+impl Context {
+    fn excute_tx(&mut self, transaction: &Transaction) -> Result<GeneralMeetingId, ExecuteError> {
+        if transaction.tx_type() != "CreateGeneralMeeting" {
+            return Err(ExecuteError::InvalidMetadata)
+        }
+        let tx: OwnTransaction = serde_cbor::from_slice(&transaction.body()).map_err(|_| ExecuteError::InvalidFormat)?;
+        tx.verify().map_err(|_| ExecuteError::InvalidSign);
+
+        let end_time = tx.tx.action.end_time;
+        let tallying = tx.tx.action.tallying_time;
+        let meeting = GeneralMeeting::new(end_time,tallying);
+        let meeting_id = &meeting.id;
+        let return_meeting_id = self.meeting_mut().create_general_meeting(meeting.clone(), meeting_id.clone()).unwrap();
+        Ok(return_meeting_id)
+    }
+}
+
+impl Stateful for Context {
+    fn set_storage(&mut self, storage: ServiceRef<dyn SubStorageAccess>) {
+        self.storage.replace(storage.unwrap_import().into_remote());
+    }
+}
+
+impl InitGenesis for Context {
+    fn begin_genesis(&mut self) {}
+
+    fn init_genesis(&mut self, config: &[u8]) {
+    }
+
+    fn end_genesis(&mut self) {}
+}
```

Comment:
If you don't do anything in functions in InitGenesis, why do you implement it?

Reply:
Here is the part that is not complete and needs our discussion because I do not have a good insight of InitGenesis.